controller/invitation.go

```diff
@@ -73,3 +73,28 @@ func (c *InvitationController) CreateInvite(ctx *app.CreateInviteInvitationConte
 
 	return ctx.Created()
 }
+
+func (c *InvitationController) AcceptInvite(ctx *app.AcceptInviteInvitationContext) error {
+	currentIdentity, err := login.LoadContextIdentityIfNotDeprovisioned(ctx, c.app)
```

Comment:
We could keep it along the same lines as 'email verification'. This endpoint should not be protected by a jwt. The user needn't be logged in. The 'code' will indirectly vouch for the validity and then the identity should be looked up using the acceptCode ( in the database ).

Reply:
So can anyone with the code accept the invitation?  The invitation repository has the following method for locating the accept code:

`InvitationRepository().FindByAcceptCode(ctx, currentIdentityID, token)`

This requires the user for which the invitation is for to be logged in, as an added level of protection.  If we wish to change this behaviour, I'll need to rewrite these methods and their tests.

Reply:
The code is *like* a secret token which should provide the necessary protection. In any case, whoever clicks on the 'accept' link, it is always going to associate the 'role' only with the identities originally intended - so we should be good. Would be great if you could re-write this to not need jwt.

Reply:
I'm still not convinced this would be a good thing - without requiring the user to be authenticated, anyone with the accept code could accept the invitation on another user's behalf.  Also, the most obvious landing page after accepting the invitation would be a redirect to the space page  (which we only support so far in the form of team and space invites) which requires the user to be authenticated anyway.  I can't see any benefits from removing the added protection of requiring the user to be already authenticated, also to change this will require changing the service method, repository find method, rewriting tests, etc.

Reply:
I've made this change, however I'm thinking that it might be useful to have two separate URLs configured for this - one for success, one for failure.  For the success, it is probably useful to include the resource ID of the space, assuming that we will be redirecting to the space page.  For a failure, we probably don't want to redirect to the space page at all.  Perhaps @rohitkrai03 could describe what would be best here?

Reply:
Thanks, I would suggest we always redirect to the same url. On success, the UI can take care of rendering the needful, to provide the user an option to navigate to the space page.

On success, we won't be passing the `error` param. So everything passed as URL params can be used to render a nice success message - and then in the same page we could have something like a "Congratulations, you've accepted the invite, CLICK HERE to visit Space X"

On failure, we *will* be passing be the `error` param. So everything passed as URL params can be used to render a nice 'did not succeed' page.

Reply:
@sbryzak , you also need to add the redirect in case of auth error / invalid jwt here

```
        currentIdentity, err := login.LoadContextIdentityIfNotDeprovisioned(ctx, c.app)
	if err != nil {
                // redirect here as well. If you return a 401/500, the user would see some plain text json  		
                return jsonapi.JSONErrorResponse(ctx, err)
	}
```

Reply:
I overlooked this one, thanks.  Fixed now.